### ISSUE TITLE
Fatal error: A void function must not return a value 

### ISSUE THREAD

**Issue by Brunty** [kahlan/kahlan#256]
I have a class that effectively looks like the following:

```php
<?php

class Outputter {

    public static function writeLine($message)
    {
        echo $message . PHP_EOL;
    }
}
```

I have a spec for it `OutputterSpec.php`:

```php
<?php

describe('Outputter', function() {
    it('writes a line', function() {
        ob_start();
        Outputter::writeLine('hi');
        $content = ob_get_contents();
        ob_end_clean();

        expect($content)->toBe('hi' . PHP_EOL);
    });
});
```

When I run this, I get the error:
```
Fatal error: A void function must not return a value in /private/var/folders/n6/abcdefghijklmnopetcetcetc/T/kahlan/Users/brunty/code/projects/cigar/src/Outputter.php on line 61
```

In that file, the method is:

```php
    public static function writeLine($message)
    {
        $__KMONKEY__0 = \Kahlan\Plugin\Monkey::patched(__NAMESPACE__, 'fwrite');
        $__KPOINTCUT_ARGS__ = func_get_args();
        $__KPOINTCUT_SELF__ = isset($this) ? $this : get_called_class();
        if ($__KPOINTCUT__ = \Kahlan\Plugin\Pointcut::before(__METHOD__, $__KPOINTCUT_SELF__, $__KPOINTCUT_ARGS__)) {
            $r = $__KPOINTCUT__($__KPOINTCUT_ARGS__, $__KPOINTCUT_SELF__);

            return $r;
        }
        $__KMONKEY__0(STDOUT, $message . PHP_EOL);
    }
```

Suggestions for testing this sort of thing? I've used the `ob_*` functions before to test such a thing in PHPUnit, wondering what the practice for testing stuff that echo in Kahlan is? Or if I'm missing something I need to do to get this working?

PHP version:
```
PHP 7.1.5 (cli) (built: May 13 2017 13:30:32) ( NTS )
```

**Comment by samsonasik** [comment]
You can do something with toEcho:

```php

            $expected = 'hi' . PHP_EOL;
            $closure = function () {
                Outputter::writeLine('hi');
            };

            expect($closure)->toEcho($expected);

```

**Comment by Brunty** [comment]
@samsonasik same error :(

**Comment by Brunty** [comment]
(but I like that format of matching echoed out content, that's nice.

**Comment by jails** [comment]
Weird, are you sure you didn't set a `void` return type on  `Outputter::writeLine()` ? Looks like the code example doesn't reflect the issue.

**Comment by Brunty** [comment]
@jails originally I did, but I took that off and it's still doing it.

**Comment by samsonasik** [comment]
@Brunty I've tested `toEcho` and working fine: https://github.com/sitrunlab/learn-kahlan/commit/15bee11aa6f03a7ae99fae1d4c7b0b888385b4ee

**Comment by jails** [comment]
@Brunty ok I think I solved your original issue https://github.com/kahlan/kahlan/commit/f2e5af5a14aa8c0983a044c7ad52503ba6277d0f so you can update your composer json to `dev/dev` if you wan't to test it.
However it doesn't make sense you are still seeing this "void" error if you removed this return type checking. Try clearing Kahlan's cached files and check that you do not have multiple versions of that file somewhere.

**Comment by Brunty** [comment]
@jails will remove the cached files - is there a command for that or shall I just `rm -rf /path/to-dir` ?

**Comment by Brunty** [comment]
Also I'll update to `dev/dev` later (I was tinkering on my lunchbreak, back to work now ;) )

**Comment by jails** [comment]
@Brunty yup `rm -rf /path/to/kahlan` is just fine.

**Comment by Brunty** [comment]
@jails just clearing the cache didn't work, so I updated to `kahlan/kahlan:dev-dev`, re-cleared the cache and it's working with the commit you added.

**Comment by jails** [comment]
Great :+1: 
Fix merged in 3.1.16 !
